docs/source/lib_builder.rst

```diff
@@ -2,37 +2,173 @@
+
+Skip Install/Update
+^^^^^^^^^^^^^^^^^^^
+
+Skip installing/updating of ESP-IDF and all components
+
+.. code-block:: bash
+
+    ./build.sh -s
+
+This option can be used if you already have the ESP-IDF and all components already in your environment.
+
+Set Arduino-ESP32 Branch
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Set which branch of arduino-esp32 to be used for compilation
+
+.. code-block:: bash
+
+    ./build.sh -A <arduino_branch>
+
+Set ESP-IDF Branch
+^^^^^^^^^^^^^^^^^^
+
+Set which branch of ESP-IDF to be used for compilation
+
+.. code-block:: bash
+
+    ./build.sh -I <idf_branch>
+
+Set the ESP-IDF Commit
+^^^^^^^^^^^^^^^^^^^^^^
+
+Set which commit of ESP-IDF to be used for compilation
+
+.. code-block:: bash
+
+    ./build.sh -i <idf_commit>
+
+Deploy
+^^^^^^
+
+Deploy the build to github arduino-esp32
+
+.. code-block:: bash
+
+    ./build.sh -d
+
+Set the Arduino-ESP32 Destination Folder
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Set the arduino-esp32 folder to copy the result to. ex. '$HOME/Arduino/hardware/espressif/esp32'
+
+.. code-block:: bash
+
+    ./build.sh -c <path>
+
+This function is used to copy the compiled libraries to the Arduino folder.
+
+Set the Target
+^^^^^^^^^^^^^^
+
+Set the build target(chip). ex. 'esp32s3'
+
+.. code-block:: bash
+
+    ./build.sh -t <target>
+
+This build command will build for the ESP32-S3 target. You can specify other targets.
+
+* esp32
+* esp32s2
+* esp32c3
+* esp32s3
 
-    ./tools/update-components.sh`
+Set Build Type
+^^^^^^^^^^^^^^
 
-- Step 4 - Run ``install-esp-idf`` installation script (if you already have an ``$IDF_PATH`` defined, it will use your local copy of the repository)::
+Set the build type. ex. 'build' to build the project and prepare for uploading to a board
 
-    ./tools/install-esp-idf.sh
+.. code-block:: bash
 
-- Step 5 - Copy the configuration (recommended) or directly edit sdkconfig using ``idf.py menuconfig``::
+    ./build.sh -b <build|menuconfig|idf_libs|copy_bootloader|mem_variant>
 
-    cp sdkconfig.esp32s2 sdkconfig
+Additional Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^
 
-- Step 6 - Build::
+Specify additional configs to be applied. ex. 'qio 80m' to compile for QIO Flash@80MHz. Requires -b
 
-    idf.py build
+.. code-block:: bash
 
-The script automates the process of building `Arduino as an ESP-IDF component <https://github.com/espressif/arduino-esp32/blob/master/docs/esp-idf_component.md>`_.
-Once it is complete, you can cherry pick the needed libraries from ``out/tools/sdk/lib``, or run ``tools/copy-to-arduino.sh`` to copy the entire built system.
-``tools/config.sh`` contains a number of variables that control the process, particularly the ``$IDF_BRANCH`` variable.  You can adjust this to try building against newer versions, but there are absolutely no guarantees that any components will work or even successfully compile against a newer IDF.
```

Comment:
This part is not "valid" anymore? Otherwise looks good.

Reply:
not valid anymore, though it will still work, there is a better way :)